rir/src/compiler/analysis/generic_static_analysis.h

```diff
@@ -197,13 +197,8 @@ class StaticAnalysis {
                         log(state);
                     }
 
-                    if (CallInstruction::CastCall(i)) {
-                        segment++;
-                        if (mergepoint[id].size() <= segment) {
-                            mergepoint[id].resize(segment + 1);
-                        }
-                        mergepoint[bb->id][segment] = state;
-                    }
+                    if (res.keepSnapshot)
+                        snapshots[bb->id].extra[i] = state;
```

Comment:
So now you can keep snapshots for any instruction, not just calls? And you pick which ones in the particular analysis?

Reply:
that's the idea. because I realized that e.g. scope analysis should also keep snapshots after analyzing force, but only if we actually dived into the promise.

otoh for many cheaper analysis it does not make sense to keep any snapshots.